osu.Game.Rulesets.Diva/Difficulty/DivaDifficultyCalculator.cs

```diff
@@ -18,9 +18,23 @@ public DivaDifficultyCalculator(Ruleset ruleset, WorkingBeatmap beatmap)
         {
         }
 
-        protected override DifficultyAttributes CreateDifficultyAttributes(IBeatmap beatmap, Mod[] mods, Skill[] skills, double clockRate)
-        {
-            return new DifficultyAttributes(mods, skills, 0);
+        protected override DifficultyAttributes CreateDifficultyAttributes(IBeatmap beatmap, Mod[] mods, Skill[] skills, double clockRate) {
+            double od = beatmap.BeatmapInfo.BaseDifficulty.OverallDifficulty;
+
+            double difficulty = 1;
+            
+            if (od > 6.0d)
+            {
+                difficulty = 4;
+            } else if (od > 4.5d)
+            {
+                difficulty = 3;
+            } else if (od > 2d)
+            {
+                difficulty = 2;
+            }
```

Comment:
please bear with me for a sec, but what is calculating `OverallDifficulty` and why can't its value simply be propagated as-is to the `DifficultyAttributes` constructor?

Reply:
OverallDifficulty is pulled from the standard map, as thats the only maps we have currently, since they are all converts

Reply:
i think it just makes it so the od of standard map and difficulty for this mode is not same

Reply:
oh right, that works until #9 is worked on
but then why not just multiply it by sth like `0.6` to keep those sweet float values?

Reply:
oh i just noticed it shows the number of keys to be used the map

Reply:
> oh i just noticed it shows the number of keys to be used in the map

Mania doesn't do that, so is it really necessary for us?

Reply:
> > oh i just noticed it shows the number of keys to be used in the map
> 
> Mania doesn't do that, so is it really necessary for us?

Its a start as not only does OD generally tell a *very* basic difficulty of a map in STD, but the more keys here makes the game exponentially harder